dataservice/api/participant/resources.py

```diff
@@ -1,114 +1,74 @@
-from flask import request
-from flask_restplus import Namespace, Resource
+from flask import abort, request
+from flask.views import MethodView
 
 from dataservice.extensions import db
-from dataservice.api.participant import models
-from dataservice.api.common.formatters import kf_response
+from dataservice.api.participant.models import Participant
+from dataservice.api.participant.schemas import ParticipantSchema
 
-description = open('dataservice/api/participant/README.md').read()
 
-participant_api = Namespace(name='participants', description=description)
-
-from dataservice.api.participant.serializers import (  # noqa
-    participant_fields,
-    participant_list,
-    participant_response
-)
-
-
-@participant_api.route('/')
-class ParticipantList(Resource):
-    @participant_api.marshal_with(participant_list)
-    def get(self):
+class ParticipantAPI(MethodView):
+    def get(self, kf_id):
         """
-        Get all participants
+        Get a participant by id
 
-        Returns paginated participants
+        Gets a participant given a Kids First id
         """
-        participants = models.Participant.query.all()
-        return kf_response(participants, 200, '{} participants'.
-                           format(len(participants)))
+        if kf_id is None:
+            return (ParticipantSchema(many=True)
+                    .jsonify(Participant.query.all()))
+        else:
+            try:
```

Comment:
Why not use one_or_none here and abort if None? In general, I think its best to avoid the catch all type try/excepts (even though there is only one line of code in the try block). Plus I think the code would look a little cleaner

Reply:
Yeah, better not to do general try/catch all. I've added the actual exception type to the exception. As for `one_or_none`, I'd really like to handle errors explicitly, in this case

Example:
```
def get_toppings(cone_id):
    icecream_cone = Cone.query.filter_by(cone_id=cone_id).one_or_none()
    if icecream_cone is None:
        abort('That cone hasnt been ordered yet!')
    if icecream_cone.toppings is None:
        return 'no fun'
    return icecream_cone.toppings
```

Compare with:
```
def get_toppings(cone_id):
    try:
        icecream_cone = Cone.query.filter_by(cone_id=cone_id).one()
    except NoResultFound:
        abort('That cone hasnt been ordered yet!')
    if icecream_cone.toppings is None:
        return 'no fun'
    return icecream_cone.toppings
```

I like the second because
1) it's easy to tell at a glance when you're handling an error vs implementing functionality
2) Exceptions are nice. We could have a general exception handler for `NoResultFound` and not even bother with the try/catch block here

Reply:
Cool, I like it 👍